cabal-install/src/Distribution/Client/ProjectConfig/Legacy.hs

```diff
@@ -2040,9 +2040,17 @@ remoteRepoSectionDescr =
     localToRemote :: LocalRepo -> RemoteRepo
     localToRemote (LocalRepo name path sharedCache) =
       (emptyRemoteRepo name)
-        { remoteRepoURI = URI "file+noindex:" Nothing path "" (if sharedCache then "#shared-cache" else "")
+        { remoteRepoURI =
+            URI
+              "file+noindex:"
+              (Just (URIAuth "" "" ""))
+              ((if isWindows then asPosixPath else id) path)
+              ""
+              (if sharedCache then "#shared-cache" else "")
         }
 
+    isWindows = buildOS == Windows
+
```

Comment:
```suggestion
        { remoteRepoURI =
            normaliseFileNoIndexURI buildOS $
              URI
                "file+noindex:"
                (Just (URIAuth "" "" ""))
                path
                ""
                (if sharedCache then "#shared-cache" else "")
        }

-- | When on Windows, we need to convert the path to be POSIX-style.
-- >>> normaliseFileNoIndexURI Windows (URI "file+noindex:" (Just (URIAuth "" "" "")) "C:\\dev\\foo" "" "")
-- file+noindex://C:/dev/foo
normaliseFileNoIndexURI :: OS -> URI -> URI
normaliseFileNoIndexURI os uri@(URI scheme auth path query fragment)
  | "file+noindex:" <- scheme
  , Windows <- os =
      URI scheme auth (asPosixPath path) query fragment
  | otherwise = uri
```

Reply:
This is what I had in mind for the doctest example. Instead of using `(Just (URIAuth "" "" ""))` could you use `(Just nullUriAuth)` or just `Nothing`?

Reply:
`Nothing` doesn't work:
```
ghci> URI "file+noindex:" Nothing "//./C:/foo.txt" "" ""
file+noindex://./C:/foo.txt
ghci> URI "file+noindex:" (Just nullURIAuth) "//./C:/foo.txt" "" ""
file+noindex:////./C:/foo.txt
```
We want the latter, because that is the only one that roundtrips:
```
ghci> fmap uriPath $ parseURI $ uriToString id (URI "file+noindex:" (Just nullURIAuth) "//./C:/foo.txt" "" "") ""
Just "//./C:/foo.txt"
ghci> fmap uriPath $ parseURI $ uriToString id (URI "file+noindex:" Nothing "//./C:/foo.txt" "" "") ""
Just "/C:/foo.txt"
```

Reply:
Used `nullURIAuth`. Thanks for the tip. I also added your function in the code-suggestion